tests/components/octoprint/test_sensor.py

```diff
@@ -170,7 +170,7 @@ async def test_sensors_printer_disconnected(hass: HomeAssistant) -> None:
 
     state = hass.states.get("sensor.octoprint_current_state")
     assert state is not None
-    assert state.state == "unavailable"
+    assert state.state == "Disconnected"
```

Comment:
Your PR writes: 

> If the printer is disconnected but OctoPrint is connected, the Current State sensor has the state "Disconnected" instead being unknown.

However, according to this test, that isn't true? It was "unavailable"? Which is the correct state IMHO.

Reply:
Sorry, I haven't described it that clear.
With this sentence I meant:
If the printer is not connected to OctoPrint, the state of the state sensor should be "Disconnected" because the printer isn't connected to OctoPrint.
Only, of course, if OctoPrint is not connected with HA the sensor should be unavailable.

This allows you to also check with the state sensor, if OctoPrint is connected to HA.
You can for example use this sensor to see if OctoPrint is online, no matter if it has a connection to the printer.

Reply:
But this sensor reflects the state of the printer, which is unavailable if it isn't connected.

Unavailable is the correct and matching state for that in Home Assistant. This entity isn't about Octoprint itself at all?

I'm confused. It seems like you try to add in a double meaning to this sensor, which is not something we should do IMHO.

Reply:
Oh, so this sensor is only about the printers state? I thought it is the state of OctoPrint and the printer. If the state was "Ready" or "Printing" I though that is the current state what OctoPrint does **with** the printer.

So this means the state should also change to "Printing" when printing from SD Card without OctoPrint doing anything, right?

Reply:
> So this means the state should also change to "Printing" when printing from SD Card without OctoPrint doing anything, right?

Correct. Octoprint reflects the state of the printer through this sensor.

Reply:
Ok. Then I understand that this doesn't really fit to this sensor. I'll close the PR.